source/ada/lsp-ada_driver.adb

```diff
@@ -46,6 +50,7 @@ procedure LSP.Ada_Driver is
 
    ALS_Dir : constant Virtual_File := Get_Home_Directory / ".als";
    Do_Exit : Boolean := False;
+   Trying_To_Finalize : Boolean := False;
```

Comment:
Do we definitely needs all this complexity? Why not to exit from the loop when server needs to be shutdown?

Reply:
Fine by me, will make the change!